hibernate-core/src/main/java/org/hibernate/sql/results/spi/ForwardOnlyScrollableResultConsumer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.sql.results.spi;
+
+import java.util.List;
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.internal.FetchingScrollableResultsImpl;
+import org.hibernate.internal.ScrollableResultsImpl;
+import org.hibernate.query.spi.ScrollableResultsImplementor;
+import org.hibernate.sql.results.graph.DomainResult;
+import org.hibernate.sql.results.graph.Fetch;
+import org.hibernate.sql.results.graph.collection.internal.EagerCollectionFetch;
+import org.hibernate.sql.results.graph.entity.EntityResult;
+import org.hibernate.sql.results.internal.RowProcessingStateStandardImpl;
+import org.hibernate.sql.results.jdbc.internal.JdbcValuesSourceProcessingStateStandardImpl;
+import org.hibernate.sql.results.jdbc.spi.JdbcValues;
+import org.hibernate.sql.results.jdbc.spi.JdbcValuesMapping;
+import org.hibernate.sql.results.jdbc.spi.JdbcValuesSourceProcessingOptions;
+
+public class ForwardOnlyScrollableResultConsumer<R> implements ResultsConsumer<ScrollableResultsImplementor<R>, R> {
+	/**
+	 * Singleton access to the standard scrollable-results consumer instance
+	 */
+	public static final ForwardOnlyScrollableResultConsumer INSTANCE = new ForwardOnlyScrollableResultConsumer();
```

Comment:
I'd prefer introducing an instance variable `boolean forwardOnly` in `ScrollableResultConsumer` and add another static final variable `ScrollableResultConsumer FORWARD_ONLY_INSTANCE = new ScrollableResultConsumer( true );` to `ScrollableResultConsumer` instead of duplicating the code. Also, this change would make some callsites megamorphic (starting with 3+ subclasses).

Reply:
yeap, this was just a draft to check if this was a viable solution.